dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java

```diff
@@ -197,49 +202,67 @@ private InputStream buildVelocity(Container container, String uuid, PageMode mod
             sb.append("#contentDetail($contentletId)");
             sb.append("#end");
 
+            if (mode == PageMode.EDIT_MODE) {
 
-                // ##Checking permission to see content
-                if (mode.showLive) {
-                    sb.append("#if($contents.doesUserHasPermission($CONTENT_INODE, 1, $user, true))");
-                }
-    
-                    // ### START BODY ###
-                    sb.append("#if($isWidget == true)");
-                        sb.append("$widgetCode");
-                    sb.append("#else");
-        
-                        for (int i = 0; i < csList.size(); i++) {
-                            ContainerStructure cs = csList.get(i);
-                            String ifelse = (i == 0) ? "if" : "elseif";
-                            sb.append("#" + ifelse + "($ContentletStructure ==\"" + cs.getStructureId() + "\")");
-                            sb.append(cs.getCode());
-                        }
-                        if (csList.size() > 0) {
-                            sb.append("#end");
-                        }
-                    // ### END BODY ###
-                    sb.append(" #end ");
-                
-               if (mode.showLive) {
-                    sb.append("#end ");
-                }
-     
+                sb.append("<div")
+                    .append(" data-dot-object=")
+                    .append("\"contentlet\"")
+                    .append(" data-dot-inode=")
+                    .append("\"$CONTENT_INODE\"")
+                    .append(" data-dot-identifier=")
+                    .append("\"$IDENTIFIER_INODE\"")
+                    .append(" data-dot-type=")
+                    .append("\"$CONTENT_TYPE\"")
+                    .append(" data-dot-lang=")
+                    .append("\"$CONTENT_LANGUAGE\"")
+                    .append(">");
 
 
+            }
+            // ##Checking permission to see content
+            if (mode.showLive) {
+                sb.append("#if($contents.doesUserHasPermission($CONTENT_INODE, 1, $user, true))");
+            }
 
+            // ### START BODY ###
+            sb.append("#if($isWidget == true)");
+                sb.append("$widgetCode");
+            sb.append("#else");
+
+                for (int i = 0; i < csList.size(); i++) {
+                    ContainerStructure cs = csList.get(i);
+                    String ifelse = (i == 0) ? "if" : "elseif";
+                    sb.append("#" + ifelse + "($ContentletStructure ==\"" + cs.getStructureId() + "\")");
+                    sb.append(cs.getCode());
+                }
+                if (csList.size() > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)